packages/cozy-harvest-lib/src/components/OAuthWindow.jsx

```diff
@@ -52,29 +53,58 @@ export class OAuthWindow extends PureComponent {
     if (typeof onCancel === 'function') onCancel()
   }
 
-  handleMessage(messageEvent) {
+  /**
+   * Handles OAuth data. OAuth data may be provided by different way:
+   * * postMessage from web apps (see handleMessage)
+   * * url changes from mobile apps
+   * @param  {string} data.key `io.cozy.accounts` id The created OAuth account
+   * @param  {string} data.oAuthStateKey key for localStorage
+   */
+  handleOAuthData(data) {
     const { konnector, onSuccess } = this.props
-
-    const { data } = messageEvent
-
     if (!checkOAuthData(konnector, data)) return
     this.setState({ succeed: true })
 
     if (typeof onSuccess !== 'function') return
+    this.setState({ succeed: true })
```

Comment:
to be removed

Reply:
Why ? To me it allows to not render the Popup anymore and trigger `Popup.componentWillUnmount`

Reply:
because we have this setState 3 lignes above?